### ISSUE TITLE
`useAwait` incorrectly warns on async functions that return promises

### ISSUE THREAD

**Issue by tylerlaprade** [biomejs/biome#3725]
### Environment information

```bash
npx biome ra
ge --linter
CLI:
  Version:                      1.8.3
  Color support:                true

Platform:
  CPU Architecture:             aarch64
  OS:                           macos

Environment:
  BIOME_LOG_DIR:                unset
  NO_COLOR:                     unset
  TERM:                         "xterm-256color"
  JS_RUNTIME_VERSION:           "v18.17.1"
  JS_RUNTIME_NAME:              "node"
  NODE_PACKAGE_MANAGER:         "npm/9.6.7"

Biome Configuration:
  Status:                       Loaded successfully
  Formatter disabled:           false
  Linter disabled:              false
  Organize imports disabled:    true
  VCS disabled:                 false

Linter:
  JavaScript enabled:           true
  JSON enabled:                 true
  CSS enabled:                  false
  Recommended:                  false
  All:                          false
  Enabled rules:
  performance/noDelete
  suspicious/noCatchAssign
  suspicious/noUnsafeNegation
  complexity/useLiteralKeys
  suspicious/useNamespaceKeyword
  complexity/noMultipleSpacesInRegularExpressionLiterals
  complexity/noUselessEmptyExport
  suspicious/useValidTypeof
  correctness/noConstantCondition
  suspicious/noDuplicateParameters
  style/useDefaultParameterLast
  complexity/noEmptyTypeParameters
  correctness/noConstructorReturn
  style/useSelfClosingElements
  style/useTemplate
  correctness/noUnusedLabels
  complexity/noUselessTernary
  correctness/noUnreachableSuper
  suspicious/noCompareNegZero
  correctness/noSwitchDeclarations
  correctness/noUnsafeOptionalChaining
  correctness/noConstAssign
  suspicious/noControlCharactersInRegex
  complexity/noUselessTypeConstraint
  style/noVar
  suspicious/noDoubleEquals
  suspicious/noRedundantUseStrict
  style/useLiteralEnumMembers
  suspicious/noGlobalIsNan
  suspicious/noEmptyInterface
  suspicious/noConstEnum
  suspicious/noSkippedTests
  suspicious/noMisleadingCharacterClass
  correctness/noPrecisionLoss
  suspicious/noRedeclare
  correctness/noStringCaseMismatch
  correctness/noSetterReturn
  correctness/noInvalidConstructorSuper
  suspicious/noImplicitAnyLet
  suspicious/noDuplicateObjectKeys
  suspicious/noUnsafeDeclarationMerging
  complexity/noUselessThisAlias
  suspicious/noFallthroughSwitchClause
  correctness/noUnreachable
  complexity/noThisInStatic
  complexity/useOptionalChain
  correctness/noInnerDeclarations
  style/noParameterAssign
  suspicious/noDuplicateCase
  complexity/useRegexLiterals
  correctness/noSelfAssign
  style/noUselessElse
  style/useShorthandFunctionType
  suspicious/noEmptyBlockStatements
  complexity/noUselessLabel
  complexity/noUselessCatch
  correctness/noUnsafeFinally
  correctness/noNonoctalDecimalEscape
  suspicious/noDuplicateTestHooks
  complexity/noStaticOnlyClass
  style/useWhile
  complexity/useArrowFunction
  style/noInferrableTypes
  complexity/useSimpleNumberKeys
  correctness/useYield
  style/useNumericLiterals
  suspicious/noImportAssign
  suspicious/noLabelVar
  correctness/noUnnecessaryContinue
  suspicious/noApproximativeNumericConstant
  suspicious/useDefaultSwitchClauseLast
  correctness/noGlobalObjectCalls
  correctness/noEmptyCharacterClassInRegex
  suspicious/noSuspiciousSemicolonInJsx
  suspicious/noSparseArray
  complexity/noBannedTypes
  suspicious/noPrototypeBuiltins
  correctness/noVoidElementsWithChildren
  style/useAsConstAssertion
  suspicious/noMisleadingInstantiator
  style/useExportType
  complexity/noUselessLoneBlockStatements
  style/noArguments
  suspicious/noGlobalAssign
  suspicious/noDebugger
  suspicious/noThenProperty
  suspicious/noCommentText
  suspicious/useGetterReturn
  suspicious/noDuplicateJsxProps
  style/useFragmentSyntax
  correctness/noEmptyPattern
  complexity/noExcessiveNestedTestSuites
  security/noDangerouslySetInnerHtmlWithChildren
  suspicious/noExtraNonNullAssertion
  correctness/noRenderReturnValue
  correctness/useExhaustiveDependencies
  security/noGlobalEval
  style/noNonNullAssertion
  style/useConst
  complexity/useFlatMap
  correctness/useHookAtTopLevel
  correctness/useIsNan
  correctness/noUnusedVariables
  suspicious/noGlobalIsFinite
  suspicious/noSelfCompare
  suspicious/noAsyncPromiseExecutor
  security/noDangerouslySetInnerHtml
  style/useNodejsImportProtocol
  suspicious/noArrayIndexKey
  complexity/noWith
  suspicious/noDuplicateClassMembers
  complexity/noExtraBooleanCast
  performance/noAccumulatingSpread
  suspicious/noConfusingLabels
  correctness/noChildrenProp
  suspicious/noFocusedTests
  correctness/noFlatMapIdentity
  correctness/useValidForDirection
  correctness/noVoidTypeReturn
  correctness/noInvalidUseBeforeDeclaration
  complexity/noUselessRename
  correctness/noInvalidNewBuiltin
  style/useNumberNamespace
  complexity/noUselessConstructor
  suspicious/useAwait
  style/useExponentiationOperator
  style/noUnusedTemplateLiteral
  complexity/noUselessSwitchCase
  style/useSingleVarDeclarator
  suspicious/noExportsInTest
  style/noCommaOperator
  suspicious/useIsArray
  complexity/noUselessFragments
  suspicious/noMisrefactoredShorthandAssign
  complexity/noForEach
  correctness/noUnusedImports
  suspicious/noClassAssign
  suspicious/noFunctionAssign

Workspace:
  Open Documents:               0
```

### Rule name

suspicious/useAwait

### Playground link

https://biomejs.dev/playground/?lintRules=all&code=YQBzAHkAbgBjACAAKAApACAAPQA%2BACAAewAKACAAIABhAHcAYQBpAHQAIABQAHIAbwBtAGkAcwBlAC4AYQBsAGwAKABbACIAZgBvAG8AIgBdAC4AbQBhAHAAKABhAHMAeQBuAGMAIAAoACkAIAA9AD4AIAB7AAoAIAAgACAAIAByAGUAdAB1AHIAbgAgAG4AZQB3ACAAUAByAG8AbQBpAHMAZQAoACkAOwAKACAAIAB9ACkAKQA7AAoAfQA7AA%3D%3D

### Expected result

It should not error. If I remove `async` to address this error, I get a complaint from ESLint:
`Functions that return promises must be async.eslint[@typescript-eslint/promise-function-async](https://typescript-eslint.io/rules/promise-function-async)`

### Code of Conduct

- [X] I agree to follow Biome's Code of Conduct

**Comment by arendjr** [comment]
It looks like Biome’s diagnostic is correct and the code in the example is incorrect. `async` functions always return a `Promise`, so by creating an explicit `Promise` instance, you’re creating a function that returns a `Promise` of a `Promise`. This makes Biome’s suggestion to not use `async` a sensible one.

ESLint’s diagnostic seems to want to take it further, but it’s a more stylistic choice: Instead of a plain function returning a `Promise` it suggests you use an `async` function returning nothing. It’s valid too, but whether it’s desirable depends more on personal taste or codebase convention, I’d say.
